### ISSUE TITLE
GFLOPS off by factor of 2 on AMD Ryzen 7 5800X 8-Core Processor

### ISSUE THREAD

**Issue by sk8ersteve** [Dr-Noob/peakperf#20]
When I run the benchmark with >8 threads, my average performance is consistently lower than the expected GFLOPS. Running peakperf with no arguments yields GFLOP of 2048 but average performance under 1200 (I have decent undervolt enabled). Specifying 8 threads changes the printed GFLOP value to 1024, and I'm seeing an average perforamance > 1100 (but less than the results for t = 16). I'm not sure if this is an issue with my cpu configuration or the test expectations.

**Comment by Dr-Noob** [comment]
Let me start by calculating your theoretical peak performance (please correct me if I'm wrong):

```
PP (SP) = 8 (cores) * 4.0 (Approx. Freq in GHz) * 2 (has FMA) * 8 (AVX2) * 2 (2 x AVX2 Units) = 1024 GFLOP/s
```

However, you seem to get 2048 GFLOP/s in some cases with 16 threads while you get around 1024 with 8 threads. There are only two possible explanations:

1. My maths are wrong.
2. There is a bug in peakperf. 

Can you please paste the complete output of peakperf with no arguments? (which will run using 16 threads).

**Comment by sk8ersteve** [comment]
Here:

------------------------------------------------------
    peakperf (https://github.com/Dr-Noob/peakperf)
------------------------------------------------------
        CPU: AMD Ryzen 7 5800X 8-Core Processor
  Microarch: Zen 3
  Benchmark: Zen 3 (AVX2)
 Iterations: 1.00e+09
      GFLOP: 2048.00
    Threads: 16

   Nº  Time(s)  GFLOP/s
    1  1.69877  1205.58 *
    2  1.72237  1189.06 *
    3  1.76517  1160.23
    4  1.71196  1196.29
    5  1.73344  1181.47
    6  1.71466  1194.41
    7  1.72736  1185.62
    8  1.71485  1194.28
    9  1.71440  1194.59
   10  1.72596  1186.59
   11  1.72048  1190.36
   12  1.71808  1192.03
------------------------------------------------------
 Average performance:      1187.50 +- 10.17 GFLOP/s
------------------------------------------------------
* - warm-up, not included in average



I've checked the code myself and added logs and it looks like you are using n_threads = 16 and op_per_it = 8. I was looking into this yesterday before I submitted the issue and assumed op_per_it/frequency was doubled to account for 2 units per core, but didn't realize that was already being accounted for in the threads being doubled. I don't know why I didn't notice this or I did but then quickly forgot, but it looks like that's the issue.
